packages/trpc/server/routers/viewer/slots/handleNotificationWhenNoSlots.test.ts

```diff
@@ -112,9 +112,357 @@ describe("(Orgs) Send admin notifications when a user has no availability", () =
+      orgDetails,
+      teamId: 123,
+    });
+
+    // Verify first notification was sent
+    expect(CalcomEmails.sendOrganizationAdminNoSlotsNotification).toHaveBeenCalledTimes(1);
+
+    // Reset call counts
+    vi.clearAllMocks();
+
+    // For the second attempt, simulate having TWO occurrences already in Redis
+    // This better simulates the real Redis state after the first notification
+    mocked.lrange.mockResolvedValueOnce(["", ""]); // Two occurrences now
+
+    await handleNotificationWhenNoSlots({
+      eventDetails,
+      orgDetails,
+      teamId: 123,
+    });
+
+    // Verify no additional notifications were sent
+    expect(CalcomEmails.sendOrganizationAdminNoSlotsNotification).not.toHaveBeenCalled();
+
+    // Verify Redis operations
+    expect(mocked.lpush).toHaveBeenCalledTimes(1); // Still records the occurrence
+  });
+
+  it("Should maintain separate notification frequencies for different event types", async () => {
+    const redisService = new RedisService();
+    const mocked = vi.mocked(redisService);
+
+    prismaMock.team.findFirst.mockResolvedValue({
+      organizationSettings: {
+        adminGetsNoSlotsNotification: true,
+      },
+    });
+
+    prismaMock.membership.findMany.mockResolvedValue([
+      {
+        user: {
+          email: "admin@test.com",
+          locale: "en",
+        },
+      },
+    ]);
+
+    const baseEventDetails = {
+      username: "user1",
+      startTime: dayjs(),
+    };
+
+    const orgDetails = {
+      currentOrgDomain: "org1",
+      isValidOrgDomain: true,
+    };
+
+    // First event type notification
+    mocked.lrange.mockResolvedValueOnce([""]); // Simulate one previous occurrence for first event
+    await handleNotificationWhenNoSlots({
+      eventDetails: { ...baseEventDetails, eventSlug: "event1" },
+      orgDetails,
+      teamId: 123,
+    });
+
+    // Verify first notification was sent
+    expect(CalcomEmails.sendOrganizationAdminNoSlotsNotification).toHaveBeenCalledTimes(1);
+
+    // Reset only the email mock, keep Redis mocks
+    vi.mocked(CalcomEmails.sendOrganizationAdminNoSlotsNotification).mockClear();
+
+    // For the second event type, also simulate one previous occurrence
+    // This needs to be a separate mock since it's a different key in Redis
+    mocked.lrange.mockResolvedValueOnce([""]); // Simulate one previous occurrence for second event
+    await handleNotificationWhenNoSlots({
+      eventDetails: { ...baseEventDetails, eventSlug: "event2" },
+      orgDetails,
+      teamId: 123,
+    });
+
+    // Verify second notification was sent (different event type)
+    expect(CalcomEmails.sendOrganizationAdminNoSlotsNotification).toHaveBeenCalledTimes(1);
+
+    // Get all lpush calls
+    const lpushCalls = mocked.lpush.mock.calls;
+    expect(lpushCalls.length).toBe(2);
+
+    // Extract the Redis keys used for each event
+    const firstEventKey = lpushCalls[0][0];
+    const secondEventKey = lpushCalls[1][0];
```

Comment:
Testing to ensure that they are called with different event types and that they keep independant counts